### PR TITLE
fix(StatusSearchLocationMenu): Refactor using new `StatusPopupMenu`

### PR DESCRIPTION
**Pull request by igor-sirotin**
Part of https://github.com/status-im/status-desktop/issues/7232
As continuation of https://github.com/status-im/status-desktop/pull/8341

### What does the PR do

* Refactor `StatusSearchLocationMenu`
* Added `ringSettings`
* Refactored `StatusMenuItemDelegate` to use `StatusSmartIdenticon`
* Removed `StatusSpellcheckingMenuItems` as unused
* Introduce `StatusMenuInstantiator` for creating menu elements from a model

⚠️  **NOTE:**
This is a PR to a temporary branch `fix/status-menu`
I'm publishing changes one by one to make the code review process easier

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [x] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [x] test changes in both light and dark theme?

### Screenshot of functionality (including design for comparison)

Снимок экрана 2022-11-24 в 16 11 07

image

Снимок экрана 2022-11-24 в 16 11 10

Снимок экрана 2022-11-24 в 16 11 13

Снимок экрана 2022-11-24 в 16 11 16

### CODE REVIEW

**Review by osmaczko — APPROVED** [review]
I haven't checked it thoroughly but at first sight it looks ok. I will have another look when you create PR for `fix/status-menu`.

File: ui/StatusQ/src/StatusQ/Components/StatusSmartIdenticon.qml [hunk 1]
```diff
@@ -44,7 +46,6 @@ Loader {
                            root.asset.bgColor
                 image.fillMode: root.asset.cropRect ? Image.PreserveAspectCrop
                                                     : Image.PreserveAspectFit
-                image.scale: root.asset.scale
```
Comment (jrainville) [inline ui/StatusQ/src/StatusQ/Components/StatusSmartIdenticon.qml:47]: This is not used?

File: ui/StatusQ/src/StatusQ/Popups/StatusMenuInstantiator.qml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        else
+            menu.addItem(object)
+    }
+
+    onObjectRemoved: {
+        if (object instanceof Menu)
+            menu.removeMenu(object)
+        else if (object instanceof Action)
+            menu.removeAction(object)
+        else // if (object instanceof MenuItem)
```
Comment (jrainville) [inline ui/StatusQ/src/StatusQ/Popups/StatusMenuInstantiator.qml:24]: Is the comment still needed?

**Review by jrainville — COMMENTED** [review]
LGTM. I' d just like one of the members of the UI cohort give a looks since there are a lot of StatusQ changes. cc @caybro

File: ui/StatusQ/src/StatusQ/Popups/StatusMenuItemDelegate.qml [hunk 3]
```diff
@@ -46,50 +91,21 @@ MenuItem {
...
-        id: indicatorLetterIdenticon
-
-        StatusLetterIdenticon {
-            width: d.assetSettings.width
-            height: d.assetSettings.height
-            color: d.assetSettings.bgColor
-            name: root.text
-            letterSize: 11
+        readonly property StatusIdenticonRingSettings defaultRingSettings: StatusIdenticonRingSettings {
+            ringPxSize: Math.max(1.5, d.assetSettings.width / 24.0)
```
Comment (PascalPrecht) [inline ui/StatusQ/src/StatusQ/Popups/StatusMenuItemDelegate.qml:96]: Not sure if worth it, but we seem to be doing the same thing here: https://github.com/status-im/status-desktop/pull/8436/files#diff-a295c8811d526e0c83e97c3c9431b689b5cce6e9ff2b0c73bcc39b003a7a18edR31 maybe it makes sense to have a little utils function for that so we ensure this doesn't get out of sync when changes are made?

File: ui/StatusQ/src/StatusQ/Popups/StatusPopupMenu.qml [hunk 4]
```diff
@@ -27,14 +43,21 @@ Menu {
 
     property StatusFontSettings fontSettings: StatusFontSettings {}
 
-    property bool hideDisabledItems: false
+    property StatusIdenticonRingSettings ringSettings: StatusIdenticonRingSettings {
+        ringPxSize: Math.max(1.5, root.assetSettings.width / 24.0)
```
Comment (PascalPrecht) [inline ui/StatusQ/src/StatusQ/Popups/StatusPopupMenu.qml:47]: Ah look, doing it a third time here

File: ui/StatusQ/src/StatusQ/Popups/StatusSearchLocationMenu.qml [hunk 5]
```diff
@@ -2,155 +2,174 @@ import QtQuick 2.14
...
+
 import StatusQ.Core 0.1
 import StatusQ.Core.Theme 0.1
-import StatusQ.Popups 0.1
+import StatusQ.Controls 0.1
 import StatusQ.Core.Utils 0.1 as StatusQUtils
 
 StatusPopupMenu {
     id: root
-    dim: false
```
Comment (PascalPrecht) [inline ui/StatusQ/src/StatusQ/Popups/StatusSearchLocationMenu.qml:12]: I assume we're getting a `dim: false` experience by other means now?

**Review by PascalPrecht — APPROVED** [review]
Skimmed through the code and judging also by screenshots, this looks good to me.
Just a minor thing to improve but I'll leave that up to you

**Pull request merged**